### PR TITLE
manual: document block expression

### PR DESCRIPTION
**Pull request by demotomohiro**
There is no explanation about block expression in the manual.rst.
So I added it.

### CODE REVIEW

**Comment by Varriount** [comment]
Isn't this just a specific case of a statement list?

**Comment by demotomohiro** [comment]
There is ``blockExpr`` in grammar.
https://nim-lang.org/docs/manual.html#syntax-grammar

This is PR that implement the block expressions.
https://github.com/nim-lang/Nim/pull/6695

Related forum thread.
https://forum.nim-lang.org/t/2201

**Comment by Varriount** [comment]
Yes, but there are also `if`, `for`, `case`, `when` and `try` expressions too.

Rather than document just block expressions, wouldn't it be preferable to expand/modify the statement list documentation to also include information on the aforementioned expressions? Alternately, a new section could be created.

**Comment by demotomohiro** [comment]
``if``, ``when``, ``case``, ``try`` expressions are already documented, but ``block``, ``for`` expressions are not.
https://nim-lang.org/docs/manual.html#statements-and-expressions-if-expression
https://nim-lang.org/docs/manual.html#statements-and-expressions-when-expression
https://nim-lang.org/docs/manual.html#statements-and-expressions-case-expression
https://nim-lang.org/docs/manual.html#exception-handling-try-expression

File: doc/manual.rst [hunk 1]
```diff
@@ -3088,6 +3088,21 @@ As seen in the above example, the case expression can also introduce side
 effects. When multiple statements are given for a branch, Nim will use
 the last expression as the result value.
 
+Block expression
+----------------
+
+A `block expression` is almost like a block statement, but it is an expression
+that use last expression under the block as the value.
```
Comment (Araq) [inline doc/manual.rst]: "uses"

**Pull request merged**